File: program/program/find-sum-of-digits-of-a-number-by-recursion/FindSumOfDigitsOfANumberByRecursion.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;

int sum_of_digits(int n)
{
    if(n==0)
    {
        return 0;
    }
    int x=n%10;
    int s=n/10;
    return x+sum_of_digits(s);
}


int main() {
	int number;
	cin>>number;
	cout<<sum_of_digits(number)<<endl;
	
	
	return 0;
}

File: program/program/find-sum-of-digits-of-a-number-by-recursion/FindSumOfDigitsOfANumberByRecursion.ts
function sumOfDigits (num: number): number {
    if(num < 10)
    {
        return num;
    }
    else
    {
        return num%10 + sumOfDigits(Math.floor(num/10));
    }
}

console.log("Sum of Digits -> 12345 is", sumOfDigits(12345))
